File: java/strategy/VisaCardPaymentStrategy.java
package strategy;

public class VisaCardPaymentStrategy implements PaymentStrategy {
    @Override
    public void processPayment(double amount) {
        System.out.println("Processing payment of visa cards...");
        System.out.println("Calculating fees of the amount" + amount + "for visa cards...");
    }
}

File: PHP/src/Behavioral/Strategy/PaymentWays/VisaCardPaymentStrategy.php
<?php

namespace DesignPatterns\Behavioral\Strategy\PaymentWays;

use DesignPatterns\Behavioral\Strategy\Interfaces\PaymentStrategy;

class VisaCardPaymentStrategy implements PaymentStrategy
{
    public function processPayment(float $amount): void
    {
        print("Processing payment of visa cards...");
        print("Calculating fees of the amount" . $amount . "for visa cards...");

    }
}
